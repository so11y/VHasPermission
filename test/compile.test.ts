import { describe, test, expect } from "vitest";
import { replaceHoistNodePlugin } from "../index";
import { readFile } from "node:fs/promises";
import { resolve, join } from "node:path";
const root = resolve(__dirname, "..");

const p = replaceHoistNodePlugin();
describe("compile test", () => {
  test("test all dir", async () => {
    const file = await readFile(join(root, "/test/fixtures.vue"));
    //@ts-ignore
    const code = p.transform(file.toString(), "./fixtures.vue");
    expect(code).toMatchSnapshot();
  });
});
