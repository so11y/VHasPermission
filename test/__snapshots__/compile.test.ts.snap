// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile test > test all dir 1`] = `
"<template>
  <HasPermission
    :permission=\\"['merchant:admin:vehicle:dispatchOrgDriver:enable']\\"
    link   >
          <TableSwitch
    
  />
          
    </HasPermission>
  <HasPermission
    :permission=\\"['merchant:admin:vehicle:dispatchOrgDriver:add']\\"
       >
          <ElButton
    
    type=\\"primary\\"
    >b</ElButton
  >
          
    </HasPermission>

  <HasPermission
    :permission=\\"[
      'merchant:admin:vehicle:dispatchOrgDriver:list',
      'merchant:admin:vehicle:dispatchOrgDriver:update',
    ]\\"
     v-if=\\"true\\"  >
          <el-dropdown-item
    
    
  >
    详情1
  </el-dropdown-item>
          
    </HasPermission>
  <HasPermission
    :permission=\\"[
      'merchant:admin:vehicle:dispatchOrgDriver:changeDispatchOrg',
    ]\\"
     v-else-if=\\"3\\"  >
          <el-dropdown-item
    
    
  >
    详情2
  </el-dropdown-item>
          
    </HasPermission>
  <HasPermission
    :permission=\\"[
      'merchant:admin:vehicle:dispatchOrgDriver:changeDispatchOrg',
    ]\\"
     v-else  >
          <el-dropdown-item
    
    
  >
    详情3
  </el-dropdown-item>
          
    </HasPermission>

  <HasPermission
    :permission=\\"[
      'merchant:admin:vehicle:dispatchOrgDriver:changeDispatchOrg',
    ]\\"
     v-if=\\"true\\"  >
          <span
    
    
    fallback=\\"无设备\\"
    class=\\"remark-text-color cursor-pointer link-text\\"
  >
    <svg-icon
      icon-class=\\"add\\"
      style=\\"margin-right: 4px\\"
      :size=\\"14\\"
      color=\\"#4772EF\\"
    />立即绑定
  </span>
          <template #fallback>无设备</template>
    </HasPermission>
</template>
"
`;
